syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -812,12 +812,14 @@ def fix_prec(self, *args, storage="auto", field_type="int100", no_wrap: bool = F
         if not kwargs.get("owner"):
             kwargs["owner"] = self.owner
 
+        # we create a copy of pointer which prevents inplace fix precision
         if self.is_wrapper:
-            self.child = self.child.fix_prec(*args, **kwargs)
+            self_copy = self.copy()
+            self_copy.child = self.child.fix_prec(*args, **kwargs)
```

Comment:
Actually I'm not 100% comfortable with copying the  tensor.
What about this?
```python
if self.is_wrapper:
    child = self.child.fix_prec(*args, **kwargs)
    if no_wrap:
        return child
    else:
        return child.wrap()
```
If this works, then I guess we should go in that direction:)

Reply:
It works and this looks better approach!!